src/fsharp/FSharp.Core/array.fs

```diff
@@ -1270,6 +1270,96 @@ namespace Microsoft.FSharp.Collections
+            checkNonNull "source" source
+            if index < 0 || index >= source.Length then invalidArg "index" "index must be within bounds of the array"
+            
+            let length = source.Length - 1
+            let result = Microsoft.FSharp.Primitives.Basics.Array.zeroCreateUnchecked length
+
+            for i in [0..length] do
+                result.[i] <-
+                    if i < index then source.[i]
+                    else source.[i + 1]
+        
+            result
+        
+        [<CompiledName("RemoveManyAt")>]
+        let removeManyAt (index: int) (count: int) (source: 'T[]) : 'T[] =
+            checkNonNull "source" source
+            if index < 0 || index > source.Length - count then invalidArg "index" "index must be within bounds of the array"
+            
+            let length = source.Length - count
+            let result = Microsoft.FSharp.Primitives.Basics.Array.zeroCreateUnchecked length
+
+            for i in [0..length] do
+                result.[i] <-
+                    if i < index then source.[i]
+                    else source.[i + count]
+            
+            result
+        
+        [<CompiledName("UpdateAt")>]
+        let updateAt (index: int) (value: 'T) (source: 'T[]) : 'T[] =
+            checkNonNull "source" source
+            if index < 0 || index >= source.Length then invalidArg "index" "index must be within bounds of the array"
+            
+            let length = source.Length
+            let result = Microsoft.FSharp.Primitives.Basics.Array.zeroCreateUnchecked length
+
+            for i in [0..length] do
+                result.[i] <-
+                    if i = index then value
+                    else source.[i]
+            result
+        
+        [<CompiledName("UpdateManyAt")>]
+        let updateManyAt (index: int) (values: seq< 'T>) (source: 'T[]) : 'T[] = 
+            checkNonNull "source" source
+            if index < 0 || index >= source.Length then invalidArg "index" "index must be within bounds of the array"
+            
+            let valuesArray = ResizeArray<_>(values)
+            let length = source.Length
+            let result = Microsoft.FSharp.Primitives.Basics.Array.zeroCreateUnchecked length
+
+            for i in [0..length] do
+                result.[i] <-
+                    if i >= index && i < index + valuesArray.Count then valuesArray.[i - index]
+                    else source.[i]
+            result
+        
+        [<CompiledName("InsertAt")>]
+        let insertAt (index: int) (value: 'T) (source: 'T[]) : 'T[] =
+            checkNonNull "source" source
+            if index < 0 || index > source.Length then invalidArg "index" "index must be within bounds of the array"
+            
+            let length = source.Length + 1
+            let result = Microsoft.FSharp.Primitives.Basics.Array.zeroCreateUnchecked length
+
+            for i in [0..length] do
```

Comment:
You delete the i value before copy to i+1

Reply:
Fixed now in the new implementation